docs/appendices/release-notes/6.0.0.rst

```diff
@@ -164,6 +164,9 @@ Breaking Changes
   inserting lots of values should be done using either bulk operations or a
   ``INSERT INTO`` in combination with a ``SELECT FROM`` and ``UNNEST``.
 
+- Set object type's depth limit to ``100`` which does not affect the tables 
+  created before :ref:`version_6.0.0`.
```

Comment:
Mentioning an increase of the limit as breaking change doesn't make sense.

I thought the change here is that it re-enables the limit with an increase.

Also - I think it ought to check tables created before 6.0.0, but _only_ if columns are added to it. And with the setting exposed as escape hatch.

Ideally we also include a rationale for it. Is this also documented elsewhere? Release notes shouldn't be the only place where a limit like this is mentioned.

Reply:
Thank you @mfussenegger 

- add a changes entry that the setting is re-enabled and increased
- add a breaking changes entry for tables < 6.0.0 that the limit could be increased with comprehension of the risks
- expose and document the setting similar to `mapping.total_fields.limit` (mention about the risk)

Reply:
>Also - I think it ought to check tables created before 6.0.0, but only if columns are added to it. And with the setting exposed as escape hatch.

Now that the setting is to be exposed, regardless of the table created versions, the column depths should be checked for newly added columns only. (For instance, we do not want to constantly be triggered by a column with large depth that was created before lowering the depth limit.) Then `writeTo()` is not a good place to check for the depths, but `addColumn()`.